### DISCUSSION TITLE
Hydration failed because the initial UI does not match what was rendered on the server.

### DISCUSSION THREAD

**Discussion by Ansh-Rathod** [vercel/next.js#41515] [Help]
Hi there i;m facing this hydration error issue. 
i have been using custom font icons and i loaded some ttf file to global.css file.

so the issue is when i remove ` <i className="icon-home"></i>` line everything works fine but when i add this tag im getting 
hydration  error :| 

Error:
![localhost_3000 - Brave 18-10-2022 13_59_16](https://user-images.githubusercontent.com/67627096/196380260-219572e2-95ab-4984-ab5f-815ec50b2d72.png)

Folder structures 
![globals css - spotify-frontend - Visual Studio Code 18-10-2022 14_02_43](https://user-images.githubusercontent.com/67627096/196380050-fde69ced-c5a7-40d9-83a7-1942fda1be39.png)

index.js
```
import type { NextPage } from "next";

import { useLogin } from "../hooks/isLoggedIn";
import Image from "next/image";

const Home: NextPage = () => {
  useLogin();

  return (
    <div className="font-ProximaRegular text-white bg-[#121212]">
      <div className="flex flex-row h-screen">
        <div className="w-[16rem] bg-black p-4">
          <div className="flex flex-row items-center">
            <Image src="/logo.jpeg" width={45} height={45} alt="logo" />
            <h1
              className="text-center uppercase mx-2 text-xl 
              tracking-wider font-ProximaBold"
            >
              Musive
            </h1>
          </div>
          <i className="icon-home"></i> <--- this line
        </div>
      </div>
    </div>
  );
};

export default Home;
```

**Comment by icyJoseph** [comment]
Weird issue.. I think you can skip the public in the font urls, like `/icomoon/fonts/icomoon.eot`, but I guess somehow an SVG or a span is placed as a child to the `i` tag?

Perhaps you could try with, https://reactjs.org/docs/dom-elements.html#suppresshydrationwarning, but only for this very particular case, though I'd try to fix it for good instead... by inlining the SVG/span or whatever is being injected on the client.

Could you show what the DOM contains (in the browser) for `icon-home`?

**Reply by Ansh-Rathod** [reply]
hey, Sorry for being late! this is what dom looks like
![localhost_3000 - Brave 19-10-2022 11_29_19 (1)](https://user-images.githubusercontent.com/67627096/196609149-be849942-99a0-452d-b992-0a2e169111bd.png)

**Reply by Ansh-Rathod** [reply]
now im getting this error after removing `<i className="icon-home"></i>` too and i'm only getting this error on index.tsx page.

**Reply by icyJoseph** [reply]
Yeah, I mean the issue is probably caused by **invalid DOM**, or code that's triggering **divergent HTML from the server to the client**, but I can't see it just yet. Could you share the logs you are getting? They are usually pretty clear about what the issue is.

If it's invalid DOM, React points out exactly where the issue is though. For divergence, it tells you that it expected something, but found something else.

If you can't tell what's different, you could also inspect the [source of the page](https://support.google.com/surveys/answer/6172725?hl=en), and try to see what's different with the Elements DOM you see in the browser console. However, I do think the logs from React should be telling you where the divergence is happening.

**Reply by icyJoseph** [reply]
Maybe the `useLogin` hook is causing some error?

**Reply by icyJoseph** [reply]
Ha, just out of curiosity @Ansh-Rathod could you try this in incognito mode? I think you might have a browser extension that's causing issues.

**Reply by Ansh-Rathod** [reply]
@icyJoseph i tried in different browser it works fine

**Reply by Ansh-Rathod** [reply]
yeah that browser extension is the problem i think

**Reply by icyJoseph** [reply]
https://github.com/darkreader/darkreader/issues/8842 Is it this one? I tried to reproduce it but couldn't, so I guess there's some kind of pattern where the extension collides with the framework. Hard to tell whose at fault here though...

I guess, issue resolved?

**Reply by Ansh-Rathod** [reply]
yes, the issue is kinda resolved if i don't use that extension. but what scenario could be in production? is there will be an issue too?

**Reply by icyJoseph** [reply]
That's hard to predict, build the application locally, and run it, with `next build` and `next start`, to see what happens. It might work, it might not... I've seen darkreader reports where the page is blank... 😞

**Reply by Ansh-Rathod** [reply]
sure, i'll try that. let's see what happens
